### ISSUE TITLE
[bug] conan lock install throws exception

### ISSUE THREAD

**Issue by snailcatcher** [conan-io/conan#8983]
### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10 1909
  * Compiler+version: gcc 8.1
  * Conan version: Conan version 1.36.0
  * Python version: Python 3.9.2

### Steps to reproduce (Include if Applicable)
```sh
conan lock install .
```
or
```sh
conan lock install conan.lock
```
or
```sh
conan lock install ./conan.lock
```

### Logs (Executed commands with output) (Include/Attach if Applicable)

```sh
Using lockfile: 'C:\...\conan.lock'
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.exception=seh
compiler.libcxx=libstdc++11
compiler.threads=posix
compiler.version=8.1
os=Windows
os_build=Windows
[options]
[build_requires]
*: mingw_installer/1.0@conan/stable, msys2/20200517
[env]

Traceback (most recent call last):
  File "c:\users\...\appdata\local\programs\python\python39\lib\site-packages\conans\client\command.py", line 2170, in run
    method(args[0][1:])
  File "c:\users\...\appdata\local\programs\python\python39\lib\site-packages\conans\client\command.py", line 1954, in lock
    self._conan.lock_install(args.lockfile, generators=args.generator, recipes=args.recipes)
  File "c:\users\...\appdata\local\programs\python\python39\lib\site-packages\conans\client\conan_api.py", line 93, in wrapper
    return f(api, *args, **kwargs)
  File "c:\users\...\appdata\local\programs\python\python39\lib\site-packages\conans\client\conan_api.py", line 1374, in lock_install
    deps_install(self.app, ref_or_path=reference, install_folder=install_folder,
  File "c:\users\...\appdata\local\programs\python\python39\lib\site-packages\conans\client\manager.py", line 57, in deps_install
    deps_graph = graph_manager.load_graph(ref_or_path, create_reference, graph_info, build_modes,
  File "c:\users\...\appdata\local\programs\python\python39\lib\site-packages\conans\client\graph\graph_manager.py", line 116, in load_graph
    root_node = self._load_root_node(reference, create_reference, profile_host, graph_lock,
  File "c:\users\...\appdata\local\programs\python\python39\lib\site-packages\conans\client\graph\graph_manager.py", line 158, in _load_root_node
    root_node = self._load_root_consumer(path, graph_lock, profile_host, root_ref)
  File "c:\users\...\appdata\local\programs\python\python39\lib\site-packages\conans\client\graph\graph_manager.py", line 170, in _load_root_consumer
    if path.endswith(".py"):
AttributeError: 'NoneType' object has no attribute 'endswith'

ERROR: 'NoneType' object has no attribute 'endswith'
```

**Comment by fittyCent** [comment]
Same here. Any luck figuring this out?

```
C:\Users\xxx\Documents\xxx\xxx\.conan>conan lock install -g visual_studio conan.lock
Using lockfile: 'C:\Users\xxx\Documents\xxx\xxx\.conan\conan.lock'
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=Visual Studio
compiler.runtime=MDd
compiler.toolset=v142
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

Traceback (most recent call last):
  File "conans\client\command.py", line 2238, in run
  File "conans\client\command.py", line 2020, in lock
  File "conans\client\conan_api.py", line 93, in wrapper
  File "conans\client\conan_api.py", line 1406, in lock_install
  File "conans\client\manager.py", line 62, in deps_install
  File "conans\client\graph\graph_manager.py", line 126, in load_graph
  File "conans\client\graph\graph_manager.py", line 171, in _load_root_node
  File "conans\client\graph\graph_manager.py", line 183, in _load_root_consumer
AttributeError: 'NoneType' object has no attribute 'endswith'
```

**Comment by memsharded** [comment]
Hi,

Sorry, I cannot reproduce this. I am trying to capture different lockfiles and doing a ``conan lock install``, but it works here.
We would need more detailed steps to reproduce this, from how the lockfile is captured in the first place.

**Comment by prvg-sso** [comment]
Hi,

having the exact same problem. Have built a conanfile.txt file on windows, and uploaded packages to my conan server. Deleted local cache, and now tries to reinstall the exact same setup.

EDIT: I converted the conanfile.txt into a python recipe, and then the problem disappeared. Perhaps this should be documented?. 



```
$> conan lock install conan_windows_debug_deps.lock
Using lockfile: 'C:\git\Repos\sso-conan-center-index\projects\PXEngine\BuildVS2022_64\locks\conan_windows_debug_deps.lock'
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=Visual Studio
compiler.cppstd=17
compiler.runtime=MTd
compiler.toolset=v143
compiler.version=17
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

Traceback (most recent call last):
  File "C:\Users\prvg\.pythonvenv\conan\lib\site-packages\conans\client\command.py", line 2216, in run
    method(args[0][1:])
  File "C:\Users\prvg\.pythonvenv\conan\lib\site-packages\conans\client\command.py", line 2018, in lock
    self._conan.lock_install(args.lockfile, generators=args.generator, recipes=args.recipes)
  File "C:\Users\prvg\.pythonvenv\conan\lib\site-packages\conans\client\conan_api.py", line 93, in wrapper
    return f(api, *args, **kwargs)
  File "C:\Users\prvg\.pythonvenv\conan\lib\site-packages\conans\client\conan_api.py", line 1407, in lock_install
    deps_install(self.app, ref_or_path=reference, install_folder=install_folder,
  File "C:\Users\prvg\.pythonvenv\conan\lib\site-packages\conans\client\manager.py", line 58, in deps_install
    deps_graph = graph_manager.load_graph(ref_or_path, create_reference, graph_info, build_modes,
  File "C:\Users\prvg\.pythonvenv\conan\lib\site-packages\conans\client\graph\graph_manager.py", line 124, in load_graph
    root_node = self._load_root_node(reference, create_reference, profile_host, graph_lock,
  File "C:\Users\prvg\.pythonvenv\conan\lib\site-packages\conans\client\graph\graph_manager.py", line 171, in _load_root_node
    root_node = self._load_root_consumer(path, graph_lock, profile_host, root_ref,
  File "C:\Users\prvg\.pythonvenv\conan\lib\site-packages\conans\client\graph\graph_manager.py", line 184, in _load_root_consumer
    if path.endswith(".py"):
AttributeError: 'NoneType' object has no attribute 'endswith'
```


conanfile.txt:
```
[requires]
glew/2.2.0@sso-sdk01/stable
glfw/3.3.7@sso-sdk01/stable
```


lockfile:
```
{
 "graph_lock": {
  "nodes": {
   "0": {
    "options": "glew:shared=False\nglfw:shared=False\nglfw:vulkan_static=False",
    "requires": [
     "1",
     "4"
    ],
    "path": "..\\..\\conan\\conanfile_windows.txt",
    "context": "host"
   },
   "1": {
    "ref": "glew/2.2.0@sso-sdk01/stable#091a57abce12f05a0e9b4b079694cdc6",
    "options": "shared=False",
    "package_id": "367d463a6988cd78d5fced97e78ced92a43c918b",
    "prev": "e461426310fcef82b50bf03be38dbe92",
    "requires": [
     "2",
     "3"
    ],
    "context": "host"
   },
   "2": {
    "ref": "opengl/system#b29589c04d6df84c110d6acb066ab653",
    "options": "",
    "package_id": "5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9",
    "prev": "b0116103765e0d967dd67f7bdcd0dd30",
    "context": "host"
   },
   "3": {
    "ref": "glu/system#ee50fc2ade0e1c0c8ebb3be3ea60e0f5",
    "options": "",
    "package_id": "5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9",
    "prev": "f9b24811cddc9ef3a9d89c24137116dc",
    "requires": [
     "2"
    ],
    "context": "host"
   },
   "4": {
    "ref": "glfw/3.3.7@sso-sdk01/stable#2b2c7fadd24b52405da5f9319596df6b",
    "options": "shared=False\nvulkan_static=False",
    "package_id": "0c4435e654bc3fe0d7f7ced16897207f5ddcf9ab",
    "prev": "7dac084388d07f97ee269362f2af1177",
    "requires": [
     "2"
    ],
    "context": "host"
   }
  },
  "revisions_enabled": true
 },
 "version": "0.4",
 "profile_host": "[settings]\narch=x86_64\narch_build=x86_64\nbuild_type=Debug\ncompiler=Visual Studio\ncompiler.cppstd=17\ncompiler.runtime=MTd\ncompiler.toolset=v143\ncompiler.version=17\nos=Windows\nos_build=Windows\n[options]\n[build_requires]\n[env]\n"
}

```
conan version 1.60.2

**Comment by memsharded** [comment]
Hi @prvg-sso 

I have managed to reproduce it. The issue is that ``conan lock install`` is more intended to install lockfiles captured with references, not with paths. Because if ``conan lock create`` uses a path, it happens that the ``conan lock install`` will require that file to be there, and cannot install only with the lockfile itself. I think it is possible to come up with a fix, but still requiring the ``conanfile`` is a bit weird, maybe you want to try to capture your lockfile with ``conan lock create <reference>`` instead of ``conan lock create <conanfilepath>``?

**Comment by memsharded** [comment]
I don't think this has enough priority to be fixed. 
The ``conan lock install`` should only be used with references, it seems that requiring the local ``conanfile`` file to be there is not only error prone and it can fail as above, but also it is risky, as reusing the same path in different machines with different conanfiles will produce unexpected behaviors.

I am closing this ticket as won't fix, as this is already gone in Conan 2.0, that was released 1 year ago, and using a reference works as expected and it was the intended usage. Thanks for all the feedback!
